### ISSUE TITLE
[Feature] List Support

### ISSUE THREAD

**Issue by nbrugger-tgm** [nbrugger-tgm/reactj#3]
It should be easy to create nested components.

For example:
A task-list has tasks in it<br>
You create a `TaskListView` and bind it to the Array of Tasks,
Then you create a `TaskView` which should be bound to an element of the List

**Comment by nbrugger-tgm** [comment]
## Event driven idea
Creating a wrapper class that catches `add` and `remove` operation on a list and binds such events to the regarding UI equivalents


**Code in mind**
Usage:
```java
ReactiveList<Task> todoList = ReactiveList.create(new ArrayList<>());
TodoListView view = new TodoListView();
view.setData(todoList);
```
UI:
```java
ReactiveListModel<Task> listModel = new ReactiveListModel();`
JList<Task> taskList = new JList(listModel.swing());
```
Binding:
```java
listModel.bind(binder);
```
`ReactiveList` would be an interface extending `Reactable`  and `List`

Non swing usage
```java
GenericPanel panel = . . . ;
ReactiveListModel<Task> listModel = new ReactiveListModel(panel::addEntry,panel::removeEntry);
```

**Comment by nbrugger-tgm** [comment]
## Rework
I am myself not satisfied with the way list support was done.
Working on a redesign at the branch `list-rework`.

Drop your ideas what to change and how to make the system better

**Comment by nbrugger-tgm** [comment]
### Dedicated Components

Creating lists is now even easier!

Extending from the `ReactiveListView` class gives you a simple way for list support by just implementing the methods
`addElementAt` `createElement` and `removeElement`!

**Comment by nbrugger-tgm** [comment]
### Examples
The branch `list-rework` contains working list examples as of the 685ba94afa7b7c8f6864f681cbbbfbac50b97541 commit

**Comment by nbrugger-tgm** [comment]
### Support Expansion

Branch `list-rework` now contains support for "basic lists" with just strings in it fd2c3f36e0f76bab8c15d879620f7562f037ec6b

#### Example

> Found in src\test\java\com\niton\reactj\examples\simplelist\StringListApp.java
```java
ReactiveList<String> someArray = ReactiveList.create(new ArrayList<>());

 JFrame frame = new JFrame();
 frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
 frame.setLayout(new GridLayout());

 SwingListView<String> view = new SwingListView<>(JLabel::new);
 view.setList(someArray);

frame.add(view.getView());
 frame.pack();
 frame.setVisible(true);

someArray.add("Can you see me?");
```
This is *fully runnable* :)

**Comment by nbrugger-tgm** [comment]
### Pre release

`2.2.2` is now available as **pre-release**

**Comment by nbrugger-tgm** [comment]
## Release
#27 (v3.0.0) contains the release of the rework

**Comment by nbrugger-tgm** [comment]
## Reopen

Due to #37 parts of the list system will be rewritten. It might just be a technical change but maybe it affects usage too

**Comment by nbrugger-tgm** [comment]
## Update

The changes to the list system are sever, therefore usage will be affected.

After the next release the list system should perform better, be way more flexible and userfriendly

**Comment by nbrugger-tgm** [comment]
## Done

List support is implemented as of `v4.0.0b7`. There are no MVC components implemented yet, but hence they do not belong to list support this feature is completed. 

List MVC components will be part of the core-mvc rework
